credentialdigger/client.py

```diff
@@ -1039,20 +1202,25 @@ def update_similar_snippets(self,
         """
 
         discoveries = self.get_discoveries(repo_url, file_name)
-        model = build_embedding_model()
-        """ Compute target snippet embedding """
-        target_snippet_embedding = compute_snippet_embedding(target_snippet,
-                                                             model)
+        # Compute target snippet embedding
+        target_embedding = self.get_embedding(snippet=target_snippet)
         n_updated_snippets = 0
-        for d in discoveries:
-            if d['state'] == 'new':
-                """ Compute snippet embedding """
-                snippet_embedding = compute_snippet_embedding(d['snippet'],
-                                                              model)
-                """ Compute similarity of target snippet and snippet """
-                similarity = compute_similarity(target_snippet_embedding,
-                                                snippet_embedding)
-                if similarity > threshold:
-                    n_updated_snippets += 1
-                    self.update_discovery(d['id'], state)
-        return n_updated_snippets
+        # If specified target snippet is not found in the embeddings table,
+        # no update is performed
+        if not target_embedding:
+            return 0
+        # We have a target embedding for the target snippet
+        else:
+            for d in discoveries:
+                if (
+                    d['state'] != state
+                    and self.get_embedding(discovery_id=d['id'])
```

Comment:
The issue with that is that we only want to retrieve embeddings that are not from the current state (and the state in not present in the discovery table). Moreover, I didn't make a get_embeddings method, because just getting a list of embeddings without any id would not be very useful, so it would need to return a list of tuples with ids and embeddings. This matching between embeddings and ids is more easily achieved using a get_embedding with the id as argument. And when I see the similarity update being performed rather instantly, even on large repos, I don't think there's a problem in making individual calls to get_embedding, but I'm happy to try otherwise